cpp/src/arrow/compute/kernels/vector_sort_test.cc

```diff
@@ -2182,5 +2185,146 @@ TEST(TestRankForFixedSizeBinary, RankFixedSizeBinary) {
       ArrayFromJSON(binary_type, R"(["aaa", "   ", "eee", null, "eee", null, "   "])"));
 }
 
+TEST(TestRankForReal, RankRealChunked) {
```

Comment:
Instead of essentially duplicating existing tests, perhaps we can generate `ChunkedArray` checks from the `Array` tests, or vice-versa?